File: IOS/v2/iosx/mediarouter/app/MediaRouteActionProvider.swift
package iOSx.mediarouter.app;.swift
import iOS.content.Context;.swift
import iOS.view.View;.swift
import iOS.view.ViewGroup;.swift
import java.lang.ref.WeakReference;.swift
public class MediaRouteActionProvider extends c8 {.swift
public final sw8 b;.swift
public final kw8 c = kw8.c;.swift
public final ov8 d = ov8.a;.swift
public cv8 e;.swift
public MediaRouteActionProvider(Context context) {.swift
super(context);.swift
this.b = sw8.d(context);.swift
new WeakReference(this);.swift
}.swift
public final boolean b() {.swift
kw8 kw8 = this.c;.swift
this.b.getClass();.swift
return sw8.e(kw8);.swift
}.swift
public final View c() {.swift
cv8 cv8 = new cv8(this.a);.swift
this.e = cv8;.swift
cv8.setCheatSheetEnabled(true);.swift
this.e.setRouteSelector(this.c);.swift
this.e.setAlwaysVisible(false);.swift
this.e.setDialogFactory(this.d);.swift
this.e.setLayoutParams(new ViewGroup.LayoutParams(-2, -1));.swift
return this.e;.swift
}.swift
public final boolean e() {.swift
cv8 cv8 = this.e;.swift
if (cv8 = null) {.swift
return cv8.d();.swift
}.swift
return false;.swift
}.swift
public final boolean g() {.swift
return true;.swift
}.swift
}.swift

File: Android/sources/androidx/mediarouter/app/MediaRouteActionProvider.java
package androidx.mediarouter.app;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import java.lang.ref.WeakReference;

public class MediaRouteActionProvider extends c8 {
    public final sw8 b;
    public final kw8 c = kw8.c;
    public final ov8 d = ov8.a;
    public cv8 e;

    public MediaRouteActionProvider(Context context) {
        super(context);
        this.b = sw8.d(context);
        new WeakReference(this);
    }

    public final boolean b() {
        kw8 kw8 = this.c;
        this.b.getClass();
        return sw8.e(kw8);
    }

    public final View c() {
        cv8 cv8 = new cv8(this.a);
        this.e = cv8;
        cv8.setCheatSheetEnabled(true);
        this.e.setRouteSelector(this.c);
        this.e.setAlwaysVisible(false);
        this.e.setDialogFactory(this.d);
        this.e.setLayoutParams(new ViewGroup.LayoutParams(-2, -1));
        return this.e;
    }

    public final boolean e() {
        cv8 cv8 = this.e;
        if (cv8 != null) {
            return cv8.d();
        }
        return false;
    }

    public final boolean g() {
        return true;
    }
}
